### PR TITLE
[Tests tidy] atoztitles service

### PR DESCRIPTION
**Pull request by tatitati**
Main changes:

- changes similar to broadcastService tests
- moved different functions under test to theirs own test class

### CODE REVIEW

File: tests/Service/AtozTitlesService/AbstractAtozTitlesServiceTest.php [hunk 1]
```diff
@@ -4,29 +4,36 @@
 
 use BBC\ProgrammesPagesService\Service\AtozTitlesService;
 use Tests\BBC\ProgrammesPagesService\AbstractServiceTest;
+use BBC\ProgrammesPagesService\Domain\Entity\AtozTitle;
```
Comment (votemike) [inline tests/Service/AtozTitlesService/AbstractAtozTitlesServiceTest.php]: Alphabetise these.

File: tests/Service/AtozTitlesService/FindAvailableTleosByFirstLetterTest.php [hunk 2]
```diff
@@ -2,54 +2,50 @@
...
 
+use BBC\ProgrammesPagesService\Domain\Entity\AtozTitle;
+
 class FindAvailableTleosByFirstLetterTest extends AbstractAtozTitlesServiceTest
 {
-    public function testFindAvailableTleosByFirstLetterDefaultPagination()
+    /**
+     * @dataProvider providerPagination
+     */
+    public function testFindAvailableTleosByFirstLetterPagination($expectedLimit, $expectedOffset, $paginationParams)
```
Comment (votemike) [inline tests/Service/AtozTitlesService/FindAvailableTleosByFirstLetterTest.php]: Typehints?

File: tests/Service/AtozTitlesService/FindAvailableTleosByFirstLetterTest.php [hunk 3]
```diff
@@ -2,54 +2,50 @@
...
+            'default pagination' => [300, 0, []],
+            'custom pagination' => [5, 10, [5, 3]],
+        ];
     }
 
-    public function testFindAvailableTleosByFirstLetterWithEmptyResult()
+    /**
+     * @dataProvider resultsProvider
+     */
+    public function testFindAvailableTleosByFirstLetterReturnRightResults($expectations, $dbResults)
```
Comment (votemike) [inline tests/Service/AtozTitlesService/FindAvailableTleosByFirstLetterTest.php]: Typehints?

File: tests/Service/AtozTitlesService/FindTleosByFirstLetterTest.php [hunk 4]
```diff
@@ -2,53 +2,49 @@
...
 
+use BBC\ProgrammesPagesService\Domain\Entity\AtozTitle;
+
 class FindTleosByFirstLetterTest extends AbstractAtozTitlesServiceTest
 {
-    public function testFindTleosByFirstLetterDefaultPagination()
+    /**
+     * @dataProvider paginationProvider
+     */
+    public function testFindTleosByFirstLetterPagination($expectedLimit, $expectedOffset, $paginationParams)
```
Comment (votemike) [inline tests/Service/AtozTitlesService/FindTleosByFirstLetterTest.php]: Typehints?

File: tests/Service/AtozTitlesService/FindTleosByFirstLetterTest.php [hunk 5]
```diff
@@ -2,53 +2,49 @@
...
+            'default pagination' => [300, 0, []],
+            'custom pagination' => [5, 10, [5, 3]],
+        ];
     }
 
-    public function testFindTleosByFirstLetterWithEmptyResult()
+    /**
+     * @dataProvider resultsProvider
+     */
+    public function testFindTleosByFirstLetterResults($expectations, $stubResults)
```
Comment (votemike) [inline tests/Service/AtozTitlesService/FindTleosByFirstLetterTest.php]: Typehints?

File: tests/Service/AtozTitlesService/AbstractAtozTitlesServiceTest.php [hunk 6]
```diff
@@ -11,23 +12,29 @@ public function setUp()
...
-    {
-        $mockAtozTitle = $this->createMock(self::ENTITY_NS . 'AtozTitle');
-        $mockAtozTitle->method('getTitle')->willReturn($entity['title']);
-        return $mockAtozTitle;
+        return new AtozTitlesService($this->mockRepository, $this->mockMapper, $this->mockCache);
     }
 
-    protected function service()
+    /**
+     * @param $tleosTitles[] $tleosTitles any model domain with getPid() function
```
Comment (votemike) [inline tests/Service/AtozTitlesService/AbstractAtozTitlesServiceTest.php]: `@param $mixed[] $tleosTitles` instead of `@param $tleosTitles[] $tleosTitles`?

Comment (tatitati) [inline tests/Service/AtozTitlesService/AbstractAtozTitlesServiceTest.php]: actually none of them , jeje
It should be 
```* @param AtozTitle[] $atozTitles ....```

Sorry!, is fixed

**Pull request merged**